Sources/ClientRuntime/PrimitiveTypeExtensions/NSLock+Extensions.swift

```diff
@@ -0,0 +1,32 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+extension NSLock {
```

Comment:
I'm hoping we can remove this in favor of using `actor`

Reply:
Seconded, see my comment above on the use of `async` while locked.

Reply:
fixed.